lnwallet/channel_test.go

```diff
@@ -574,6 +578,234 @@ func TestSimpleAddSettleWorkflow(t *testing.T) {
 	}
 }
 
+// TestCheckCommitTxSize checks that estimation size of commitment
+// transaction with some degree of error corresponds to the actual size.
+func TestCheckCommitTxSize(t *testing.T) {
+	checkSize := func(channel *LightningChannel, count int) {
+		// Due to variable size of the signatures (71-73) we may have
+		// an estimation error.
+		BaseCommitmentTxSizeEstimationError := 4
+
+		commitTx, err := channel.getSignedCommitTx()
+		if err != nil {
+			t.Fatalf("unable to initiate alice force close: %v", err)
+		}
+
+		actualCost := blockchain.GetMsgTxCost(commitTx)
+		estimatedCost := estimateCommitTxCost(count, false)
+
+		diff := int(estimatedCost - actualCost)
+		if 0 > diff || BaseCommitmentTxSizeEstimationError < diff {
+			t.Fatalf("estimation is wrong")
+		}
+
+	}
+
+	createHTLC := func(i int) (*lnwire.HTLCAddRequest, [32]byte) {
+		preimage := bytes.Repeat([]byte{byte(i)}, 32)
+		paymentHash := fastsha256.Sum256(preimage)
+
+		var returnPreimage [32]byte
+		copy(returnPreimage[:], preimage)
+
+		return &lnwire.HTLCAddRequest{
+			RedemptionHashes: [][32]byte{paymentHash},
+			Amount:           lnwire.CreditsAmount(1e7),
+			Expiry:           uint32(5),
+		}, returnPreimage
+	}
+
+	signNextCommitment := func(alice, bob *LightningChannel) {
```

Comment:
The addition of this local helper function isn't necessary since the function `forceStateTransition` already exists and performs the exact same duties.